engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1695,6 +1607,42 @@ public void run()
             }
             endWindowStatsMap.put(shb.getNodeId(), endWindowStats);
 
+            if (!oper.getInputs().isEmpty()) {
+              long latency = Long.MAX_VALUE;
```

Comment:
@davidyan74 : Consider following scenario
For window W1, stats are received in following order A,C,B and Latency (A) < Latency(B) but since B arrived later than C, C adds A as it's slowest upstream.

For Window W2, stats are received in B,A,C order and new Latency(A) > new Latency(B), now again C adds A as it's slowest upstream. 

B is never added as slowest upstream for C in this scenario. 

Again it depends on how accurate you want to show the latency values.